packages/bsky/src/api/blob-resolver.ts

```diff
@@ -0,0 +1,79 @@
+import { pipeline, Readable } from 'stream'
+import express from 'express'
+import createError from 'http-errors'
+import axios, { AxiosError } from 'axios'
+import { CID } from 'multiformats/cid'
+import { ensureValidDid } from '@atproto/identifier'
+import { VerifyCidTransform } from '@atproto/common'
+import { NoResolveDidError } from '@atproto/did-resolver'
+import AppContext from '../context'
+import { httpLogger as log } from '../logger'
+
+// Resolve and verify blob from its origin host
+
+export const createRouter = (ctx: AppContext): express.Router => {
+  const router = express.Router()
+
+  router.get('/blob/:did/:cid', async function (req, res, next) {
+    try {
+      const { did, cid: cidStr } = req.params
+      try {
+        ensureValidDid(did)
+      } catch (err) {
+        return next(createError(400, 'Invalid did'))
+      }
+      let cid: CID
+      try {
+        cid = CID.parse(cidStr)
+      } catch (err) {
+        return next(createError(400, 'Invalid cid'))
+      }
+
+      const { pds } = await ctx.didResolver.resolveAtpData(did) // @TODO cache did info
+      const getBlob = await axios.get(`${pds}/xrpc/com.atproto.sync.getBlob`, {
+        params: { did, cid: cidStr },
+        decompress: true,
+        responseType: 'stream',
+        timeout: 2000, // 2sec of inactivity on the connection
+      })
+
+      const imageStream: Readable = getBlob.data
+      const verifyCid = new VerifyCidTransform(cid)
+
+      // Send chunked response, destroying stream early (before
+      // closing chunk) if the bytes don't match the expected cid.
+      res.statusCode = 200
+      res.setHeader(
+        'content-type',
+        getBlob.headers['content-type'] || 'application/octet-stream',
```

Comment:
I checked the `getBlob` route on the PDS & I think it's always giving `application/octet-stream` 🤔 

May wanna sneak a change in for that here if we're feeling up for it

Reply:
I forget, does imgproxy need a hint as to the content type? Or it can just take raw bytes & figure out the rest just fine?

Reply:
It should figure it out just fine without the content-type hint 👍 I'll make that change though, looks pretty straightforward.